presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -81,24 +87,81 @@ public static Slice charToVarcharCast(@LiteralParameter("x") Long x, @LiteralPar
     @ScalarOperator(OperatorType.SATURATED_FLOOR_CAST)
     @SqlType("char(y)")
     @LiteralParameters({"x", "y"})
-    // This function returns Char(y) value that is smaller than original Varchar(x) value. However, it is not necessarily the largest
-    // Char(y) value that is smaller than the original Varchar(x) value. This is fine though for usage in TupleDomainTranslator.
-    public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") Long y, @SqlType("varchar(x)") Slice slice)
+    public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") long y, @SqlType("varchar(x)") Slice slice)
     {
-        Slice trimmedSlice = trimTrailingSpaces(slice);
-        int trimmedTextLength = countCodePoints(trimmedSlice);
-        int numberOfTrailingSpaces = slice.length() - trimmedSlice.length();
+        List<Integer> codePoints = new ArrayList<>(toCodePoints(slice));
 
         // if Varchar(x) value length (including spaces) is greater than y, we can just truncate it
-        if (trimmedTextLength + numberOfTrailingSpaces >= y) {
-            return truncateToLength(trimmedSlice, y.intValue());
+        if (codePoints.size() >= y) {
+            // char(y) slice representation doesn't contain trailing spaces
+            trimTrailingSpaces(codePoints);
```

Comment:
The comment line above ("char(y) slice representation doesn't contain trailing spaces") was supposed to explain that. Can I improve it somehow?